### DISCUSSION TITLE
Missing index after using tool

### DISCUSSION THREAD

**Discussion by TantumNTatari** [agronholm/sqlacodegen#171] [Q&A]
After using this tool to reflect a table in my database, the declarative model that was produced was missing an index. I was just wondering if anyone might know why this is the case:

The actual database, using the '\d+ users' command
Table "public.users"
      Column       |           Type           | Collation | Nullable |              Default              | Storage  | Stats target | Description 
-------------------+--------------------------+-----------+----------+-----------------------------------+----------+--------------+-------------
 id                | integer                  |           | not null | nextval('users_id_seq'::regclass) | plain    |              | 
 email             | text                     |           | not null |                                   | extended |              | 
 password          | bytea                    |           | not null |                                   | extended |              | 
Indexes:
    "users_pkey" PRIMARY KEY, btree (id)
    "users_email_unique_idx" UNIQUE, btree (lower(email))
    "users_username_password_idx" btree (email, password)

However, the tool would only show the following:
Table "public.users"
      Column       |           Type           | Collation | Nullable |              Default              | Storage  | Stats target | Description 
-------------------+--------------------------+-----------+----------+-----------------------------------+----------+--------------+-------------
 id                | integer                  |           | not null | nextval('users_id_seq'::regclass) | plain    |              | 
 email             | text                     |           | not null |                                   | extended |              | 
 password          | bytea                    |           | not null |                                   | extended |              | 
Indexes:
    "users_pkey" PRIMARY KEY, btree (id)
    "users_username_password_idx" btree (email, password)

Thanks.

**Comment by agronholm** [comment]
Could you post the generated code? Which version did you use?

**Reply by TantumNTatari** [reply]
This is using version 2.3.0:

```
class User(Base):
    __tablename__ = 'users'
    __table_args__ = (
        Index('users_username_password_idx', 'email', 'password'),
    )

    id = Column(Integer, primary_key=True, server_default=text("nextval('users_id_seq'::regclass)"))
    email = Column(Text, nullable=False)
    password = Column(LargeBinary, nullable=False)
```

**Reply by agronholm** [reply]
Does v3.0.0b2 exhibit the same problem?

**Reply by TantumNTatari** [reply]
I have just tried with v3.0.0b2 and I believe I got the same output.

**Reply by agronholm** [reply]
Which version of SQLAlchemy was this on?

**Reply by TantumNTatari** [reply]
I believe it is SQLAlchemy version 1.4.26.

**Reply by agronholm** [reply]
Ok, thanks. I'll try to reproduce this locally and see what's what.

**Reply by agronholm** [reply]
Did you not get this warning at the top?
```
SAWarning: Skipped unsupported reflection of expression-based index users_email_unique_idx
```

**Reply by TantumNTatari** [reply]
I see it now... I deeply apologize for not examining this thoroughly enough. Thank you for your support. I will continue to figure out a solution for my problem.

**Reply by agronholm** [reply]
You just need to add it to `__table_args__` manually. Nothing to it.

**Reply by soypedroamaya** [reply]
> I see it now... I deeply apologize for not examining this thoroughly enough. Thank you for your support. I will continue to figure out a solution for my problem.

Hello, I was wondering how you solved this, I am having the same problem:

sqlacodegen/main.py:47: SAWarning: Skipped unsupported reflection of expression-based index users_instance_id_email_idx
   metadata.reflect(engine, args.schema, not args.noviews, tables)

**Reply by agronholm** [reply]
If SQLAlchemy can't support some form of reflection, there's nothing that sqlacodegen can do. You'll have to add such indexes by hand to the generated code.
